sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/namedExpressions.scala

```diff
@@ -578,6 +578,11 @@ object FileSourceGeneratedMetadataAttribute {
 
   val FILE_SOURCE_GENERATED_METADATA_COL_ATTR_KEY = "__file_source_generated_metadata_col"
 
+  // We keep generated metadata attributes nullability configurable here:
+  // 1. Before passing to readers, we create generated metadata attributes as nullable;
+  //    Because, for row_index, the readers do not consider the column required.
+  // 2. When applying the projection, we change the nullability back to not-nullable;
+  // See [[FileSourceStrategy.scala]] for more information
```

Comment:
For regular comments, should better use backticks instead of brackets (that are for proper Scala docs)

Reply:
Thanks! Addressed